src/cr/cube/cube.py

```diff
@@ -784,20 +796,21 @@ class _UnweightedCountMeasure(_BaseMeasure):
 
     @lazyproperty
     def _flat_values(self):
-        """tuple of int counts before weighting."""
-        if (
-            self._cube_dict["result"]["measures"]
-            .get("valid_count_unweighted", {})
-            .get("data")
-        ):
-            # ---If valid_count are expressed in the cube dict, returns its data.
-            # ---This condition can happen in case of numeric array cube response.
-            # ---Under this circumstances the numeric array measures will contain the
-            # ---mean measure and a valid count measure for the unweighted counts.
-            return tuple(
-                self._cube_dict["result"]["measures"]["valid_count_unweighted"]["data"]
-            )
-        return tuple(self._cube_dict["result"]["counts"])
+        """np.ndarray of np.float64 counts before weighting or None, if unavailable.
+
+        Use floats to avoid int overflow bugs and so we can use nan.
+        """
+        result = self._cube_dict["result"]
+
+        # ---If valid_count are expressed in the cube dict, returns its data.
+        # ---This condition can happen in case of numeric array cube response.
+        # ---Under this circumstances the numeric array measures will contain the
+        # ---mean measure and a valid count measure for the unweighted counts.
+        valid_counts = result["measures"].get("valid_count_unweighted", {}).get("data")
+        if valid_counts:
+            return np.array(valid_counts, dtype=np.float64)
+
+        return np.array(result["counts"], dtype=np.float64)
```

Comment:
It's a very fine point, but just in case you're interested, I would generally phrase this as:

```python
return (
    np.array(valid_counts, dtype=np.float64)
    if valid counts
    else np.array(result["counts"], dtype=np.float64)
)
```

I'm not entirely sure why I like this better, except that as a reader you don't have to reason about two separate statements and maybe any unhandled "fall-through" cases or something; like it's a straight-line execution path and says "okay, we're returning a value now, there's no two ways about it, it's just a question of what that value is going to be.

Anyway do it the way you like best, just wanted to share my perspective on that.